src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -142,11 +150,19 @@ protected TestResult parse(String expandedTestResults, AbstractBuild build, Laun
     @Override
     public void perform(Run build, FilePath workspace, Launcher launcher,
             TaskListener listener) throws InterruptedException, IOException {
+        TestResultAction action = parseAndAttach(this, null, build, workspace, launcher, listener);
+
+        if (action != null && action.getResult().getFailCount() > 0)
+            build.setResult(Result.UNSTABLE);
```

Comment:
This doesn't feel right to me -- don't people tend to handle the way test results map to build results in vastly different ways, generally?